src/unstract/sdk/index.py

```diff
@@ -287,6 +274,18 @@ def index(
                 documents.append(document)
             self.tool.stream_log(f"Number of documents: {len(documents)}")
 
+            if doc_id_found:
```

Comment:
@Deepak-Kesavan if the doc_id_found is what happens? We will still index right? May be add a comment here to make it clear.

Reply:
if doc_id_found we delete and index, else we just index alone.